### PR TITLE
feat(menuCategories): add  endpoints to handle menu categories

### LINKED ISSUE #10: Create menu categories

### LINKED ISSUE #11: Read menu categories

### LINKED ISSUE #12: Update menu categories

### PR DESCRIPTION
**Pull request by jcmexdev**
I finished the menu categories module 
## Taks
- Migration to create menu_categories table
- List all Categories ✅
- Create Categories ✅
- Get one category ✅
- Update Category ✅

## Helpers
I added the endpoint to FastOrder postman team under **MenuCategories** collection , so you can test it.

### CODE REVIEW

**Comment by AlanDavd** [comment]
LGTM 👍

File: src/modules/menu-category/menu-category.service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  async create(newMenuCategory: CreateMenuCategoryDto) {
+    const menuCategory = await this._menuCategoryRepository.findOne({ name: newMenuCategory.name });
+
+    if (menuCategory) throw new ConflictException('menu_category_name_already_exists');
+
+    const theMenuCategory = plainToClass(MenuCategory, newMenuCategory);
+    const timestamps = new Date();
+
+    theMenuCategory.createdAt = timestamps;
```
Comment (jlazo) [inline src/modules/menu-category/menu-category.service.ts]: Tanto el createdAt como el updatedAt, debido al decorator CreateDateColumn y UpdateDateColumn, no es necesario setearle una fecha, ya lo hace solo

File: src/modules/menu-category/menu-category.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return plainToClass(ResponseMenuCategoryDto, await theMenuCategory.save());
+  }
+
+  async update(id: number, updateMenuCategoryDto: UpdateMenuCategoryDto) {
+    const menuCategory = await this._menuCategoryRepository.findOne(id);
+
+    if (!menuCategory) throw new ConflictException('menu_category_was_not_found');
+
+    menuCategory.name = updateMenuCategoryDto.name;
+    menuCategory.updatedAt = new Date();
```
Comment (jlazo) [inline src/modules/menu-category/menu-category.service.ts]: Igual que en el comentario previo

**Review by jlazo — CHANGES_REQUESTED** [review]

File: src/modules/menu-category/menu-category.service.ts (hunk 1 above)
Comment (jcmexdev) [reply src/modules/menu-category/menu-category.service.ts]: oh, i see, i'm going to remove that,it isn't necessary  ✅

File: src/modules/menu-category/menu-category.entity.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { BaseEntity, Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn } from "typeorm";
+
+@Entity('menu_categories')
+export class MenuCategory extends BaseEntity {
+  @PrimaryGeneratedColumn('increment')
+  id: number;
+  @Column({ type: 'varchar', name: 'name', length: 50 })
+  name: string;
+  @Column({ name: 'author_id', nullable: true })
```
Comment (jlazo) [inline src/modules/menu-category/menu-category.entity.ts]: Here, you must put the column as nullable: false

**Review by jlazo — CHANGES_REQUESTED** [review]

File: src/modules/menu-category/menu-category.entity.ts (hunk 3 above)
Comment (jcmexdev) [reply src/modules/menu-category/menu-category.entity.ts]: OK, i get it 🤖

**Review by jlazo — APPROVED** [review]

**Pull request merged**